include/tracelogging/TraceLoggingProvider.h

```diff
@@ -1853,41 +1883,47 @@ inline void _tlgCppInit1DescSeqUtf32(struct lttngh_DataDesc &desc, ctype const *
 }
 
 template <class ctype>
-inline void _tlgCppInit1DescWchar(struct lttngh_DataDesc &desc, ctype const &value) _tlg_NOEXCEPT _tlg_INLINE_ATTRIBUTES;
+inline void _tlgCppInit1DescWchar(struct lttngh_DataDesc &desc, ctype const &value, _tlgCharBuf<wchar_t>&& buf = _tlgCharBuf<wchar_t>()) _tlg_NOEXCEPT _tlg_INLINE_ATTRIBUTES;
```

Comment:
The default parameter thing is a [weird] trick to create a buffer where I can turn the char into a nul-terminated string. The buffer has to exist on the caller's stack space so that it doesn't go away when this function returns. This function is used in code looking something like:
`_tlgCppInit1DescWchar(&dataDesc[3], ch), EventWrite(dataDesc);`
Note the use of a comma, which means temporaries survive until after EventWrite returns, which means this trick works.